app/main.py

```diff
@@ -1,34 +1,157 @@
+from typing import Optional
+
+
 class Deck:
-    def __init__(self, row, column, is_alive=True):
-        pass
+    def __init__(self, row: int, col: int, is_alive: bool = True) -> None:
+        self.row = row
+        self.col = col
+        self.is_alive = is_alive
+
+    def __repr__(self) -> str:
+        return f"Deck(row={self.row}, col={self.col}, " \
+               f"is_alive={self.is_alive})"
 
 
 class Ship:
-    def __init__(self, start, end, is_drowned=False):
-        # Create decks and save them to a list `self.decks`
-        pass
+    def __init__(
+            self,
+            start: tuple[int, int],
+            end: tuple[int, int],
+            is_drowned: bool = False
+    ) -> None:
+        self.is_drowned = is_drowned
+        self._initialize_decks(start, end)
+
+    def _initialize_decks(
+            self,
+            start: tuple[int, int],
+            end: tuple[int, int]
+    ) -> None:
+        if start[1] != end[1]:
+            self.decks = [Deck(start[0], index)
+                          for index in range(start[1], end[1] + 1)]
+        elif start[0] != end[0]:
+            self.decks = [Deck(index, end[1])
+                          for index in range(start[0], end[0] + 1)]
+        else:
+            self.decks = [Deck(start[0], end[1])
+                          for index in range(start[0], end[0] + 1)]
+
+    def location(self) -> list:
+        return [(deck.row, deck.column) for deck in self.decks]
+
+    def get_deck(self, row: int, col: int) -> Optional[Deck]:
+        return next((deck for deck in self.decks
+                     if deck.row == row and deck.col == col), None)
```

Comment:
In this way better to use also filter:
```suggestion
        return next(
            filter(
                lambda: deck: deck.row == row and deck.col == col, 
                self.decks
            ), 
            None
        )
```